### PR TITLE
Re-sync Chipyard and FireSim

### PR DESCRIPTION
**Pull request by davidbiancolin**
There was little Chipyard dev over the summer so i'd be putting this off but it is necessary prerequisite for attempting a chisel 3.5 bump.  In bumping CY i could deploy the new GlobalResetCondition annotation.  As part of this effort I:

- updated the AnnotationWiringTransfrom to let children define the behavior of the transform under 0 source or 0 sink annotations -- which may or may not be acceptable depending on the application. Firrtl's WiringTransform errors in either case.
- added extra floorplan constraints in aws-fpga to improve build robustness. 

#### Related PRs / Issues

ucb-bar/chipyard#976
firesim/aws-fpga-firesim#45

#### UI / API Impact

No change on FireSim's side. 

#### Verilog / AGFI Compatibility

Some chipyard RTL changes. AGFIs have been regenerated.

### Contributor Checklist
- [x] Did you set dev as the base branch?
- [x] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you state the UI / API impact?
- [x] Did you specify the Verilog / AGFI compatibility impact?

- [ ] (If applicable) Did you regenerate and publicly share default AGFIs?

- [ ] (If applicable) Did you mark the PR as "Please Backport"?
- [ ] (On merge) Did you update release notes in the dev-to-master PR ?

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

**Review by jerryz123 — APPROVED** [review]

File: sim/src/main/cc/firesim/firesim_top.cc [hunk 1]
```diff
@@ -568,9 +568,6 @@ void firesim_top_t::run() {
     fprintf(stderr, "Commencing simulation.\n");
     record_start_times();
 
-    // Assert reset T=0 -> 50
```
Comment (abejgonzalez) [inline sim/src/main/cc/firesim/firesim_top.cc:571]: Why was this removed?

**Review by abejgonzalez — APPROVED** [review]
LGTM

File: sim/src/main/cc/firesim/firesim_top.cc (hunk 1 above)
Comment (davidbiancolin) [reply sim/src/main/cc/firesim/firesim_top.cc:571]: The resetPulseBridge now drives the target reset -- this resolves the reset deadlock bug you saw at one point, because this function can no-longer prevent execution of the simulator's main loop.
